src/main/java/com/github/mc1arke/sonarqube/plugin/ce/pullrequest/PullRequestPostAnalysisTask.java

```diff
@@ -148,19 +152,57 @@ public void finished(Context context) {
 
         String commitId = revision.get();
 
+        ScannerContext scannerContext = projectAnalysis.getScannerContext().getProperties().size() > 0
+                ? projectAnalysis.getScannerContext()
+                : GetScannerContext(projectAnalysis.getCeTask().getId());
+
         AnalysisDetails analysisDetails =
                 new AnalysisDetails(new AnalysisDetails.BranchDetails(optionalBranchName.get(), commitId),
                                     postAnalysisIssueVisitor, qualityGate,
                                     new AnalysisDetails.MeasuresHolder(metricRepository, measureRepository,
                                                                        treeRootHolder), analysis,
                                     projectAnalysis.getProject(), configuration, server.getPublicRootUrl(),
-                                    projectAnalysis.getScannerContext());
+                                    scannerContext);
 
         PullRequestBuildStatusDecorator pullRequestDecorator = optionalPullRequestDecorator.get();
         LOGGER.info("using pull request decorator" + pullRequestDecorator.name());
         pullRequestDecorator.decorateQualityGateStatus(analysisDetails, almSettingDto, projectAlmSettingDto);
     }
 
+    public ScannerContext GetScannerContext(String ceTaskId)
```

Comment:
1. method can be made private
1. first letter of method name should be lowercased

Reply:
Resolved